File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/ColumnsSimpleDemo/ColumnsSimpleDemoCars.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "ColumnsSimpleDemoCars.h"
#include "ColumnsSimpleDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxCustomData"
#pragma link "cxData"
#pragma link "cxEdit"
#pragma link "cxFilter"
#pragma link "cxGraphics"
#pragma link "cxGrid"
#pragma link "cxGridBandedTableView"
#pragma link "cxGridCustomTableView"
#pragma link "cxGridCustomView"
#pragma link "cxGridDBBandedTableView"
#pragma link "cxGridLevel"
#pragma link "cxGridTableView"
#pragma link "cxStyles"
#pragma link "cxBlobEdit"
#pragma link "cxDataStorage"
#pragma link "cxDBData"
#pragma link "cxHyperLinkEdit"
#pragma link "cxImage"
#pragma link "cxImageComboBox"
#pragma link "cxMemo"
#pragma link "cxRadioGroup"
#pragma resource "*.dfm"
TColumnSimpleDemoCarsForm *ColumnSimpleDemoCarsForm;
//---------------------------------------------------------------------------
__fastcall TColumnSimpleDemoCarsForm::TColumnSimpleDemoCarsForm(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TColumnSimpleDemoCarsForm::bvCarsTopRecordIndexChanged(
      TObject *Sender)
{
  bvCars->Controller->FocusedRecordIndex = bvCars->Controller->TopRecordIndex;
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/ColumnsSimpleDemo/ColumnsSimpleDemoCars.pas
unit ColumnsSimpleDemoCars;

interface

uses
  Windows, Messages, Classes, Graphics, Controls, Forms,
  Dialogs, cxGridCustomTableView, cxGridTableView, cxGridBandedTableView,
  cxGridDBBandedTableView, cxControls, cxGridCustomView,
  cxGridLevel, cxGrid, Grids, DBGrids, ExtCtrls, cxStyles, cxCustomData,
  cxGraphics, cxFilter, cxData, cxEdit, cxClasses, cxDataStorage, DB,
  cxDBData, cxImage, cxBlobEdit, cxRadioGroup, cxImageComboBox, cxMemo,
  cxHyperLinkEdit, cxLookAndFeels, cxLookAndFeelPainters, cxNavigator, CarsDataForGrid;

type
  TColumnSimpleDemoCarsForm = class(TForm)
    pnlCarInfo: TPanel;
    GridCars: TcxGrid;
    bvCars: TcxGridDBBandedTableView;
    bvCarsID: TcxGridDBBandedColumn;
    bvCarsTrademark: TcxGridDBBandedColumn;
    bvCarsModel: TcxGridDBBandedColumn;
    bvCarsPrice: TcxGridDBBandedColumn;
    bvCarsPicture: TcxGridDBBandedColumn;
    bvCarsLargePicture: TcxGridDBBandedColumn;
    bvCarsCategory: TcxGridDBBandedColumn;
    bvCarsHP: TcxGridDBBandedColumn;
    bvCarsCyl: TcxGridDBBandedColumn;
    bvCarsTransmissSpeedCount: TcxGridDBBandedColumn;
    bvCarsTransmissAutomatic: TcxGridDBBandedColumn;
    bvCarsMPG_City: TcxGridDBBandedColumn;
    bvCarsMPG_Highway: TcxGridDBBandedColumn;
    bvCarsDescription: TcxGridDBBandedColumn;
    bvCarsHyperlink: TcxGridDBBandedColumn;
    lvCars: TcxGridLevel;
    procedure bvCarsTopRecordIndexChanged(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  ColumnSimpleDemoCarsForm: TColumnSimpleDemoCarsForm;

implementation

uses ColumnsSimpleDemoData;

{$R *.dfm}

procedure TColumnSimpleDemoCarsForm.bvCarsTopRecordIndexChanged(
  Sender: TObject);
begin
  bvCars.Controller.FocusedRecordIndex := bvCars.Controller.TopRecordIndex;
end;

end.
